### ISSUE TITLE
How do I continue training with new dataset on a pretrained model like transfer learning(Domain adaptation) 

### ISSUE THREAD

**Issue by valivetiravichandra** [OpenNMT/OpenNMT-py#728]
Hello,
Suppose I have a pre trained model for English to German and want to continue training with pretrained weights on new dataset(English to German but medical domain or bank). Both datasets i.e train and new domain data being different. How do I do this?

**Comment by valivetiravichandra** [comment]
Its very interesting how this is done. Its like we have transfer learning in image recognition but I'm trying to imagine how it would work for language translation like adding words to already pretrained model. In this way we even save gpu time and resources.
If anyone have an idea please comment here.

**Comment by mojesty** [comment]
What you (as well as I) possibly need is described [here](https://arxiv.org/abs/1604.02201).
There people try to freeze some parameters of a model learned in large corpus (called parental model) and learn the rest initialized randomly. They report good results, but here's the question: how to freeze parameters. While it is definitely possible (via changing optimizer), I can't find any simple way to do it except modifying the source.

**Comment by lchunleo** [comment]
http://opennmt.net/OpenNMT/training/retraining/

Does this somehow solve the retraining questions?

I am unsure about the statement that train from does not.do so for the vocabulary.

**Comment by valivetiravichandra** [comment]
I don't think we can add new vocabulary and train the pretrained model.

**Comment by lchunleo** [comment]
So does it meant if i have a new dataset, with the current model design, i would need to append them to the old dataset and retrain everything again? 

sorry am a newbie in DL NLP.

**Comment by lchunleo** [comment]
i have tried to use the retraining of train_from and noted that my result with my new training dataset (but same validation data) has not really improve my model accuracy. it tends to hover around a constant value and never beyond. Why is it so?

And i would like to check how do i use the tensorboard to view the training? i have run up the tensorboard but there is no output despite specifying the tensorboard directory during the training invocation command

**Comment by vince62s** [comment]
closing for now, open a new issue with specific point if needed. thanks.

**Comment by echan00** [comment]
also interested in doing this
